File: Day_1/puzzle2.py
increase: int = 1

int_data: list[int] = []

with open("input.txt", 'r') as f:
    data = f.read().split()
    for l in range(0, len(data)):
        int_data.append(int(data[l]))

    total = [sum(int_data[i:i+3]) for i in range(1, len(int_data))]
    print(total)
    f.close()

for i in range(0, len(total)):
    if total[i] > total[i-1] and total[i] != total[0]:
        increase += 1
print(f"increase of {increase}")

File: Day_2/puzzle2.lua
local open = io.open

local values = {}
local direction = {}

local calculate_direction = {}

calculate_direction["forward"] = 0
calculate_direction["depth"] = 0
calculate_direction["aim"] = 0

local function read_file(path)
    local file = open(path, "rb")
    if not file then
        return nil
    end
    local content = file:read("*a")
    file:close()
    return content
end

local function calculation( dir, distance )
        if dir == "forward" then
            calculate_direction["forward"] = calculate_direction["forward"] + distance
            print(calculate_direction["forward"])
            calculate_direction["depth"] = calculate_direction["depth"] + (calculate_direction["aim"] * distance)
            print(calculate_direction["depth"])
        elseif dir == "down" then
            calculate_direction["aim"] = calculate_direction["aim"] + distance
            print(calculate_direction["aim"])
        elseif dir == "up" then
            calculate_direction["aim"] = calculate_direction["aim"] - distance
            print(calculate_direction["aim"])
        end
    end


local filecontent = read_file("input.txt")

for token in string.gmatch(filecontent, "[^%s]+") do
    if tonumber(token) ~= nil then
        table.insert(values, tonumber(token))
    elseif tonumber(token) == nil then
        table.insert(direction, token)
    end
end

for i = 1, #values do
    --print("direction: ", direction[i])
    --print("distance: ", values[i])
    calculation(direction[i], values[i])
end

print("forward value: ", calculate_direction["forward"])
print("depth value: ", calculate_direction["depth"])
print("total: ", calculate_direction["forward"] * calculate_direction["depth"])
